library/core/src/num/nonzero.rs

```diff
@@ -2214,4 +2232,28 @@ mod verify {
     nonzero_check!(u64, core::num::NonZeroU64, nonzero_check_new_unchecked_for_u64);
     nonzero_check!(u128, core::num::NonZeroU128, nonzero_check_new_unchecked_for_u128);
     nonzero_check!(usize, core::num::NonZeroUsize, nonzero_check_new_unchecked_for_usize);
-}
+
+    #[kani::proof_for_contract(i8::unchecked_mul)]
+    fn nonzero_unchecked_mul() {
+        let x: NonZeroI8 = kani::any();
+        let y: NonZeroI8 = kani::any();
+
+        // Check the precondition to assume no overflow
+        let (result, overflow) = x.get().overflowing_mul(y.get());
+        kani::assume(!overflow); // Ensure the multiplication does not overflow
```

Comment:
```suggestion
```
As a rule of thumb, we shouldn't have any `kani::assume`s in a `proof_for_contract`. The only exception is if you need them for performance reasons, e.g., you need to restrict the range of integers that Kani checks so that verification finishes in a reasonable amount of time. In this case, this assumption just repeats your contract precondition, which makes it redundant; Kani expands contract preconditions to `kani::assume`s under the hood.